### ISSUE TITLE
Query entities with non-unique field by latest DAOs

### ISSUE THREAD

**Issue by waynerv** [Kong/kong#5275]
### Summary

Is there any way for the new DAO framework to query entities with a non-unique field?
I'm seeking something just like removed `crud_helpers.find_by_id_or_field`  acting a normal filter querying but only found generated query method for foreign key or unique field in new DAOs.

It's also confusing that there isn't any document mentioned that `kong.db.<entity>` wil generate `each_for_<foreign_key>` or `select_by_<unique_field>` methods automately.
Took a lot of time to read source code for this.

kong version: v1.4

**Comment by bungle** [comment]
There is `select_by_cache_key` which can take multiple columns which make it unique. The DAO's need to add column to db on such case. One such entity is plugins:
https://github.com/Kong/kong/blob/master/kong/db/schema/entities/plugins.lua#L8

See also:
https://github.com/Kong/kong/blob/master/kong/db/migrations/core/001_14_to_15.lua#L74-L80

And:
https://github.com/Kong/kong/blob/master/kong/db/migrations/core/001_14_to_15.lua#L303

**Comment by bungle** [comment]
TL;DR:

```
crud_helpers.find_by_id_or_field
```

Is almost impossible to implement efficiently on Cassandra. Thus our new DAO is in some respects more limited than the old one. Old one had a lot of functionality that was hard to optimize.

**Comment by dndx** [comment]
As @bungle mentioned, this is something we can not do efficiently in Cassandra, which means we can not implement it right now. We may reconsider this decision in the next major release of Kong.
